lib/db/lowlevel.go

```diff
@@ -590,6 +627,122 @@ func (db *Lowlevel) gcBlocks() error {
 	return db.Compact()
 }
 
+func (db *Lowlevel) gcDeleted() error {
+	// The deleted file GC finds items that are deleted and older than
+	// deletedGCAge (default 180 days). If we are in sync on this item (no
+	// need entry) we remove it from the database, along with the
+	// corresponding remote entries.
+	//
+	// If the remote side doesn't GC deleted files then the entry will at
+	// some point pop up again. Ideally this should therefore be
+	// symmetrically configured.
+
+	if deletedGCAge <= 0 {
+		// <= 0 is explicitly disabled
+		return nil
+	}
+	if deletedGCAge < deletedGCMinAge {
+		// an explicit but not reasonable value, we adjust upwards a bit.
+		deletedGCAge = deletedGCMinAge
+	}
+
+	db.gcMut.Lock()
+	defer db.gcMut.Unlock()
+
+	t, err := db.newReadWriteTransaction()
+	if err != nil {
+		return err
+	}
+	defer t.Release()
+
+	cutoff := time.Now().Add(-deletedGCAge)
+
+	it, err := db.NewPrefixIterator([]byte{KeyTypeSequence})
+	if err != nil {
+		return err
+	}
+	for it.Next() {
```

Comment:
I'm iterating the sequence list because my initial thought was that I could stop once a too-new deleted item is reached. However there may be deleted items with times in the future because broken clocks and then we'd never gc, and I didn't change it to some other iteration afterwards...